solidity/test/TestPhasedEscrow.js

```diff
@@ -249,6 +268,157 @@ describe("PhasedEscrow", () => {
         })
       })
     })
+
+    describe("when withdrawing to a staker rewards escrow beneficiary", () => {
+      const baseBalance = 200000000
+      let stakingContract
+      let operatorContract
+      let rewardsBeneficiary
+      let phasedEscrow
+
+      before(async () => {
+        const contracts = await initContracts(
+          contract.fromArtifact("TokenStaking"),
+          contract.fromArtifact("KeepRandomBeaconService"),
+          contract.fromArtifact("KeepRandomBeaconServiceImplV1"),
+          contract.fromArtifact("KeepRandomBeaconOperatorBeaconRewardsStub")
+        )
+
+        stakingContract = contracts.stakingContract
+        operatorContract = contracts.operatorContract
+
+        phasedEscrow = await PhasedEscrow.new(token.address, {from: owner})
+        const amount = web3.utils.toBN(baseBalance)
+        await token.transfer(phasedEscrow.address, amount, {from: owner})
+      })
+
+      beforeEach(createSnapshot)
+      afterEach(restoreSnapshot)
+
+      describe("BeaconBackportRewardsEscrowBeneficiary", () => {
```

Comment:
Hmm... I see why you did that but I think we complicated the structure this way. Instead of:

```
PhasedEscrow
  when withdrawing to a CurveRewardsEscrowBeneficiary
  when withdrawing to a staker rewards escrow beneficiary
    BeaconBackportRewardsEscrowBeneficiary
    BeaconRewardsEscrowBeneficiary
    ECDSABackportRewardsEscrowBeneficiary
    ECDSARewardsEscrowBeneficiary
```

I think we should have:
```
PhasedEscrow
  when withdrawing to a CurveRewardsEscrowBeneficiary
  when withdrawing to a BeaconBackportRewardsEscrowBeneficiary
  when withdrawing to a BeaconRewardsEscrowBeneficiary
  when withdrawing to a ECDSABackportRewardsEscrowBeneficiary
  when withdrawing to a ECDSARewardsEscrowBeneficiary
```

Or - IMO even better - if we decide for a parant contract (previous comment), do:
```
PhasedEscrow
  when withdrawing to a CurveRewardsEscrowBeneficiary
  when withdrawing to a StakerRewardsBeneficiary
```

Reply:
We have a possibility to test against the actual `*BeaconRewards` contract that I set up in this test. But `*ECDSARewards` is a mock. If we decide to test only against the parent contract, then either we sacrifice the possibility of testing against real `*BeaconRewards` or not testing against a mock for `*ECDSARewards`. What do you think if we implement a parent contract as you suggested above, but having a second structure as you posted here?

Reply:
Yeah, I think it's a good move. Parent contract with the code but separate `describe` for each beneficiary:
```
PhasedEscrow
  when withdrawing to a CurveRewardsEscrowBeneficiary
  when withdrawing to a BeaconBackportRewardsEscrowBeneficiary
  when withdrawing to a BeaconRewardsEscrowBeneficiary
  when withdrawing to a ECDSABackportRewardsEscrowBeneficiary
  when withdrawing to a ECDSARewardsEscrowBeneficiary
```

Reply:
Done.